src/main/java/de/taimos/pipeline/aws/S3UploadStep.java

```diff
@@ -487,6 +535,17 @@ public Void invoke(File localFile, VirtualChannel channel) throws IOException, I
 					request.withMetadata(metas);
 				}
 
+				//add tags
+				if(tags != null){
+					List<Tag> tagList = new ArrayList<Tag>();
```

Comment:
Using a lambda expression, this could be a oneliner.

Reply:
Agreed

Reply:
Done.